### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by migos**
* update libs
* remove unused and incompatible libs
* update gradle
* update java version
* create config
* update setup.sh
* replace loco

when these changes are merged we can proceed with creating a branch for jetpack compose support

### CODE REVIEW

File: app/src/main/kotlin/com/tailoredapps/androidapptemplate/MyApp.kt [hunk 1]
```diff
@@ -38,13 +38,10 @@ class MyApp : Application() {
         AndroidThreeTen.init(this)
 
         startKoin {
-            androidLogger(Level.INFO)
+            androidLogger(Level.ERROR)
```
Comment (daniel-diamir) [inline app/src/main/kotlin/com/tailoredapps/androidapptemplate/MyApp.kt]: debug for debug builds?

File: app/src/main/kotlin/com/tailoredapps/androidapptemplate/detail/DetailView.kt [hunk 2]
```diff
@@ -35,16 +37,16 @@ class DetailView(
...
     private val binding by viewBinding(FragmentDetailBinding::bind)
     private val navController by lazy(::findNavController)
     private val viewModel by viewModel<DetailViewModel>()
 
-    init {
-        launchOnViewLifecycleStartCancelOnViewLifecycleStop {
-
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        repeatOnLifeCycleStart {
```
Comment (daniel-diamir) [inline app/src/main/kotlin/com/tailoredapps/androidapptemplate/detail/DetailView.kt]:
```suggestion
        launchWhenStartedCancelWhenStopped {
```
Thats how we call it in our project.
The former name was bit long but I think it is nice to now it is cancelled on stop or what do you think?

If not we should add it to the kdoc of the extension when it is cancelled ("cancels when it falls below that state" is a bit _confusing_ at times)

File: base-ui/src/main/kotlin/com/tailoredapps/androidapptemplate/base/ui/ControllerViewModel.kt [hunk 3]
```diff
@@ -28,12 +28,13 @@ abstract class ControllerViewModel<Action, State> : ViewModel() {
...
     fun dispatch(action: Action) = controller.dispatch(action)
-    val currentState: State get() = controller.currentState
+    val currentState: State get() = controller.state.value
     val state: Flow<State> get() = controller.state
 
     @CallSuper
     override fun onCleared() {
         super.onCleared()
-        AppWatcher.objectWatcher.watch(this, "${this::class.java.simpleName}.onCleared")
+        AppWatcher.objectWatcher.expectWeaklyReachable(this,
```
Comment (daniel-diamir) [inline base-ui/src/main/kotlin/com/tailoredapps/androidapptemplate/base/ui/ControllerViewModel.kt]: maybe separate lines and named arguments.

File: base-ui/src/main/kotlin/com/tailoredapps/androidapptemplate/base/ui/LifecycleOwnerExt.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Convenience function to for [repeatOnLifecycle] with [Lifecycle.State.CREATED]
+ */
+fun Fragment.repeatOnLifeCycleCreated(scope: CoroutineScope.() -> Unit) {
+    repeatOnLifeCycle(Lifecycle.State.CREATED, scope)
+}
+
+/**
+ * Convenience function to for [repeatOnLifecycle] with [Lifecycle.State.RESUMED]
```
Comment (daniel-diamir) [inline base-ui/src/main/kotlin/com/tailoredapps/androidapptemplate/base/ui/LifecycleOwnerExt.kt]:
```suggestion
 * Convenience function for [repeatOnLifecycle] with [Lifecycle.State.RESUMED]
```

or

```suggestion
 * Extension for [repeatOnLifecycle] with [Lifecycle.State.RESUMED]
```

File: build.gradle [hunk 5]
```diff
@@ -1,32 +1,22 @@
...
-        jcenter()
-        maven { url "https://plugins.gradle.org/m2/" }
+        mavenCentral()
     }
 
     dependencies {
         classpath(Libs.com_android_tools_build_gradle)
         classpath(Libs.kotlin_gradle_plugin)
-        classpath(Libs.de_fayard_refreshversions_gradle_plugin)
-        classpath(Libs.easylauncher)
```
Comment (daniel-diamir) [inline build.gradle:18]: 👍

File: settings.gradle [hunk 6]
```diff
@@ -1 +1,10 @@
+plugins {
+    // See https://jmfayard.github.io/refreshVersions
+    id 'de.fayard.refreshVersions' version '0.40.1'
+}
+
+refreshVersions {
+    enableBuildSrcLibs() // <-- Add this
```
Comment (daniel-diamir) [inline settings.gradle]: what does the comment mean?

**Review by daniel-diamir — DISMISSED** [review]
LGTM - just some minor nitpicks and comments :)
thx for the update 🥇

File: buildSrc/src/main/kotlin/Config.kt [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import org.gradle.api.JavaVersion
+
+object Config {
+
+    @JvmStatic
```
Comment (AlexGrafl) [inline buildSrc/src/main/kotlin/Config.kt:5]: i suppose this is necessary when used in groovy gradle files, so maybe its time to move to kotlin gradle?

**Review by AlexGrafl — COMMENTED** [review]
Generally LGTM.
Pls add a section in the README how to use the new "refreshVersions" version

**Review by FlowMo7 — DISMISSED** [review]
LGTM! :rocket:

File: app/src/main/kotlin/com/tailoredapps/androidapptemplate/MyApp.kt [hunk 8]
```diff
@@ -38,7 +37,8 @@ class MyApp : Application() {
         AndroidThreeTen.init(this)
 
         startKoin {
-            androidLogger(Level.ERROR)
+            val logLevel = if (BuildConfig.DEBUG) Level.INFO else Level.ERROR
```
Comment (daniel-diamir) [inline app/src/main/kotlin/com/tailoredapps/androidapptemplate/MyApp.kt:40]: Level.DEBUG?

File: core/build.gradle [hunk 9]
```diff
@@ -1,7 +1,7 @@
 apply from: rootProject.file("gradle/library-module.gradle")
 
 dependencies {
-    api(Libs.control_core)
+//    api(Libs.control_core)
```
Comment (daniel-diamir) [inline core/build.gradle]: remove?

**Review by daniel-diamir — DISMISSED** [review]
```kotlin

          _____            _____                _____                    _____          
         /\    \          /\    \              /\    \                  /\    \         
        /::\____\        /::\    \            /::\    \                /::\____\        
       /:::/    /       /::::\    \           \:::\    \              /::::|   |        
      /:::/    /       /::::::\    \           \:::\    \            /:::::|   |        
     /:::/    /       /:::/\:::\    \           \:::\    \          /::::::|   |        
    /:::/    /       /:::/  \:::\    \           \:::\    \        /:::/|::|   |        
   /:::/    /       /:::/    \:::\    \          /::::\    \      /:::/ |::|   |        
  /:::/    /       /:::/    / \:::\    \        /::::::\    \    /:::/  |::|___|______  
 /:::/    /       /:::/    /   \:::\ ___\      /:::/\:::\    \  /:::/   |::::::::\    \ 
/:::/____/       /:::/____/  ___\:::|    |    /:::/  \:::\____\/:::/    |:::::::::\____\
\:::\    \       \:::\    \ /\  /:::|____|   /:::/    \::/    /\::/    / ~~~~~/:::/    /
 \:::\    \       \:::\    /::\ \::/    /   /:::/    / \/____/  \/____/      /:::/    / 
  \:::\    \       \:::\   \:::\ \/____/   /:::/    /                       /:::/    /  
   \:::\    \       \:::\   \:::\____\    /:::/    /                       /:::/    /   
    \:::\    \       \:::\  /:::/    /    \::/    /                       /:::/    /    
     \:::\    \       \:::\/:::/    /      \/____/                       /:::/    /     
      \:::\    \       \::::::/    /                                    /:::/    /      
       \:::\____\       \::::/    /                                    /:::/    /       
        \::/    /        \::/____/                                     \::/    /        
         \/____/                                                        \/____/         
                                                                                        
```

**Review by FlowMo7 — DISMISSED** [review]

File: app/src/main/kotlin/com/tailoredapps/androidapptemplate/MyApp.kt (hunk 8 above)
Comment (migos) [reply app/src/main/kotlin/com/tailoredapps/androidapptemplate/MyApp.kt:40]: I think `Level.DEBUG` spams to much logs and `Level.INFO` is fine for most cases. If more logging is needed the dev could change it.

Comment (daniel-diamir) [reply app/src/main/kotlin/com/tailoredapps/androidapptemplate/MyApp.kt:40]: I get the spam point, but the package only should only be visible per default also.
🧐 don't you always use debug as default for you to log? so everyone that does that would have to change to see the logs, right?

**Review by FlowMo7 — APPROVED** [review]

**Review by daniel-diamir — APPROVED** [review]

**Pull request merged**